cotk/dataloader/dataloader.py

```diff
@@ -169,36 +240,65 @@ def chain_allvocab(dic):
 		print("vocab list length = %d" % len(vocab_list))
 
 		word2id = {w: i for i, w in enumerate(vocab_list)}
+
 		def line2id(line):
-			return ([self.go_id] + \
+			return [self.go_id] + \
 					list(map(lambda word: word2id[word] if word in word2id else self.unk_id, line)) \
-					+ [self.eos_id])[:max_sent_length]
+					+ [self.eos_id]
+
+		def to_id(element, type_):
+			type_ = type_.lower().strip()
+			if type_ == 'label':
+				return element
+			if type_ == 'sentence':
+				return line2id(element)
+			if type_ == 'session':
+				return [line2id(sentence) for sentence in element]
+
+		def cut(element, type_):
+			if type_ == 'label':
+				return element
+			if type_ == 'sentence':
+				return element[: max_sent_length]
+			if type_ == 'session':
+				return [sentence[:max_sent_length] for sentence in element[:max_turn_length]]
 
 		data = {}
 		data_size = {}
 		for key in self.key_name:
 			data[key] = {}
-			for data_key in data_keys:
-				data[key][data_key] = list(map(line2id, origin_data[key][data_key]))
+			for data_key, type_ in data_format:
+				origin_data[key][data_key] = [to_id(element, type_) for element in origin_data[key][data_key]]
+				data[key][data_key] = [cut(element, type_) for element in origin_data[key][data_key]]
 				if key not in data_size:
 					data_size[key] = len(data[key][data_key])
 				elif data_size[key] != len(data[key][data_key]):
-					raise RuntimeError("The number of lines of input %s.txt should be able divisible by %d" % (key, len(data_keys)))
+					raise RuntimeError(
+						"The number of lines of input %s.txt should be able divisible by %d" % (key, len(data_format)))
```

Comment:
We have sessions, so we should say    The data of input %s.txt contains different numbers of fields